### PR TITLE
create filter bug fix and payload change

### PR DESCRIPTION
**Pull request by AbijithS-aot**
# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE

# Changes

fixed ui bugs in create filter and the order of the payload changed based on different conditions
# How has the change been tested?

# Screenshots (if applicable)

image
image
image

### CODE REVIEW

**Review by abhilash-aot — APPROVED** [review]

File: forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { useSelector } from "react-redux";
 import Drawer from "@material-ui/core/Drawer";
 import Button from "@material-ui/core/Button";
 import List from "@material-ui/core/List";
 import Divider from "@material-ui/core/Divider";
 import { saveFilters } from "../../../../apiManager/services/bpmTaskServices";
+import { ACCESSIBLE_FOR_ALL_GROUPS, PRIVATE_ONLY_YOU, SPECIFIC_USER_OR_GROUP } from "../../../../constants/taskConstants";
```
Comment (shuhaib-aot) [inline forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js:8]: PRIVATE_ONLY is better than PRIVATE_ONLY_YOU

File: forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js [hunk 2]
```diff
@@ -13,36 +14,59 @@ export default function CreateNewFilterDrawer() {
...
+  const userName = useSelector((state) => state.user?.userDetail?.preferred_username);
+  const [variables,setVariables] = useState([]);
+  const [inputValues, setInputValues] = useState([{ name: '', label: '' }]);
+
+  useEffect(()=>{
+    setVariables(() => {
+      if (inputValues.length === 1 && inputValues[0].name === '' && inputValues[0].label === '') {
+        return [];
+      } else {
+        return inputValues.map((row) => ({
```
Comment (shuhaib-aot) [inline forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js:32]: why need to do here map ? could you please explain

File: forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js [hunk 3]
```diff
@@ -53,13 +77,13 @@ export default function CreateNewFilterDrawer() {
...
         setIncludeAssignedTasks("");
         setPermissions("");
         setIdentifierId("");
         setSelectUserGroupIcon("");
         setSpecificUserGroup("");
-        setRows([defaultRow]);
+        setInputValues([{ name: '', label: '' }]);
       })
       .catch((error) => {
         console.error("error", error);
```
Comment (shuhaib-aot) [inline forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js:89]: toast the error

**Review by shuhaib-aot — CHANGES_REQUESTED** [review]

File: forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js (hunk 1 above)
Comment (AbijithS-aot) [reply forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js:8]: that can be changed

File: forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js (hunk 3 above)
Comment (AbijithS-aot) [reply forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js:89]: ok

File: forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js (hunk 2 above)
Comment (AbijithS-aot) [reply forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js:32]: here map is used to set all input values to variable state

**Pull request merged**